tfe/data_source_oauth_clients_test.go

```diff
@@ -0,0 +1,68 @@
+package tfe
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccTFEOAuthClientsDataSource_basic(t *testing.T) {
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			if GITHUB_TOKEN == "" {
+				t.Skip("Please set GITHUB_TOKEN to run this test")
+			}
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{Destroy: false,
+				PreventPostDestroyRefresh: true,
+				Config:                    testAccTFEOAuthClientsDataSourceConfig(rInt),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"data.tfe_oauth_clients.foobar", "oauth_clients.#", "2"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_oauth_clients.foobar", "oauth_clients.0.service_provider", "github"),
+					resource.TestCheckResourceAttr(
```

Comment:
We should probably sanity check a few more attributes!

Reply:
Alternatively, we've typically have `full` and `basic` acceptance tests, where `TestAcc*_full` validates all attributes and `TestAcc*_basic` validates *required* attributes, or in this case attributes that must be present.